base/sparse/cholmod.jl

```diff
@@ -790,9 +838,37 @@ function sparse{Ti}(A::Sparse{Complex{Float64},Ti}) # Notice! Cannot be type sta
     end
     return convert(Hermitian{Complex{Float64},SparseMatrixCSC{Complex{Float64},Ti}}, A)
 end
-sparse(L::Factor)   = sparse(Sparse(L))
+function sparse(F::Factor)
+    s = unsafe_load(F.p)
+    if bool(s.is_ll)
+        L = Sparse(F)
+        A = sparse(L*L')
+    else
+        LD = sparse(F[:LD])
+        L, d = getLd!(LD)
+        A = scale(L, d)*L'
+    end
+    p = get_perm(F)
+    if p != [1:s.n;]
+        pinv = Array(Int, length(p))
+        for k = 1:length(p)
+            pinv[p[k]] = k
+        end
+        A = A[pinv,pinv]
```

Comment:
I'm simply glad you're reading it at all, no apologies needed.

To be clear, a `FactorComponent` is any type of factor: it can represent just `:L` or `:D` (meaning, it also represents single factors) or `:PtL` or `:LD` or ...

Rather than mushing them together into a single type, I think it would be fine to have separated factors using an expanded type hierarchy: `L = cholfact[:L]; P = cholfact[:P]`. But I do wonder whether ready-made combinations are still desirable, in much the same way that a `Cholesky` is a nice convenience that guards against dumb mistakes like `x = L\(L'\b)` (instead of `L'\(L\b)` as it should be). Or stated another way, what advantages do you see that our `cholfact` types offer that's different from `L, p = chol(A)`?